### PR TITLE
Bugfix & improvement of Cyclone V SoC DK ethernet

### PR DESCRIPTION
**Pull request by benkyd**
Bugfix & improvement of Cyclone V SoC DK ethernet

Issue involved overflow of a timeout counter and the driver instance was not the emac-index.

Signed-off-by: Benjamin Kyd <benjamin.kyd@intel.com>

### CODE REVIEW

File: drivers/ethernet/phy_cyclonev.c [hunk 1]
```diff
@@ -47,8 +47,8 @@
 #define PHY_RESTART_AUTONEGOTIATION BIT(9)
 
 /* Basic Status Register */
-#define PHY_BSR		      BIT(0)
-#define PHY_AUTOCAP	      BIT(30) /* Auto-negotiation capability */
+#define PHY_BSR		      (1)
```
Comment (teburd) [inline drivers/ethernet/phy_cyclonev.c:50]: This looks a bit odd next to the others, if its a bit flag at bit 0 BIT(0) was correct

**Review by teburd — DISMISSED** [review]
Looks fine to me, the change from BIT(0) to (1) is a bit odd looking in the context of the commit and might be worth changing back if it really is a bit flag at Bit 0.

File: drivers/ethernet/phy_cyclonev.c (hunk 1 above)
Comment (benkyd) [reply drivers/ethernet/phy_cyclonev.c]: Changed that, i see how it can look odd, also changed the PHY_BSR flag. PHY_CR will stay as an integer though as i would need to check what it represents

**Review by teburd — DISMISSED** [review]

**Comment by benkyd** [comment]
@teburd Apologies for the push once more.. the compliance check failed again at a spot i thought i amended... Should be good now once CI passes. Sorry again

**Review by teburd — DISMISSED** [review]

**Comment by benkyd** [comment]
@teburd do you think you could request a review from the others.. they seem to have been automatically removed.

File: drivers/ethernet/eth_cyclonev.c [hunk 2]
```diff
@@ -315,6 +315,7 @@ static int eth_cyclonev_set_config(const struct device *dev, enum ethernet_confi
 				   const struct ethernet_config *config)
 {
 	struct eth_cyclonev_priv *p = dev->data;
+	const struct eth_cyclonev_config *cv_config = dev->data;
```
Comment (MaureenHelm) [inline drivers/ethernet/eth_cyclonev.c:318]: This looks wrong. It should probably be:
```suggestion
	const struct eth_cyclonev_config *cv_config = dev->config;
```

File: dts/arm/intel_socfpga_std/socfpga.dtsi [hunk 3]
```diff
@@ -136,16 +136,16 @@
...
 		};
 
 		gmac1: ethernet@ff702000 {
 			compatible = "snps,ethernet-cyclonev";
 			reg = <0xff702000 0x2000>;
 			interrupts = <0 120 4 IRQ_DEFAULT_PRIORITY>;
-			local-mac-address = [e6 88 2d 82 1f d4];
-			status = "okay";
+			emac-index = <1>;
+			status = "disabled";
```
Comment (MaureenHelm) [inline dts/arm/intel_socfpga_std/socfpga.dtsi:148]: The commit message doesn't explain this change. Why are you removing `local-mac-address` and changing `status`?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: dts/arm/intel_socfpga_std/socfpga.dtsi (hunk 3 above)
Comment (benkyd) [reply dts/arm/intel_socfpga_std/socfpga.dtsi:148]: The "local-mac-address" property is moved from dts/arm/intel_socfpga_std/socfpga.dtsi to boards/arm/cyclonev_socdk/cyclonev_socdk.dts, since this value is dependent on the board and not universal to "intel_socfpga"

File: drivers/ethernet/eth_cyclonev.c [hunk 4]
```diff
@@ -1142,30 +1143,30 @@ const struct ethernet_api eth_cyclonev_api = {.iface_api.init = eth_cyclonev_ifa
...
+					&eth_cyclonev_api, \
+					NET_ETH_MTU); \
+ \
+	static void eth_cyclonev_##inst##_irq_config(void) \
+	{ \
+		IRQ_CONNECT(DT_INST_IRQN(inst), \
+			    DT_INST_IRQ(inst, priority), eth_cyclonev_isr, \
+			    DEVICE_DT_INST_GET(inst), \
+			    0); \
+		irq_enable(DT_INST_IRQN(inst)); \
```
Comment (MaureenHelm) [inline drivers/ethernet/eth_cyclonev.c:1170]: It's hard to see if there's a functional change hidden in these, what look like mostly whitespace changes. Please extract the whitespace changes to a separate commit

File: drivers/ethernet/eth_cyclonev.c [hunk 5]
```diff
@@ -1145,13 +1145,13 @@ const struct ethernet_api eth_cyclonev_api = {.iface_api.init = eth_cyclonev_ifa
...
  \
-	static struct eth_cyclonev_priv eth_cyclonev_##inst##_data; \
+	static struct eth_cyclonev_priv eth_cyclonev_##inst##_data;	\
 	static void eth_cyclonev_##inst##_irq_config(void); \
  \
 	static const struct eth_cyclonev_config eth_cyclonev_##inst##_cfg = { \
 			.base = (uint8_t *)(DT_INST_REG_ADDR(inst)), \
 			.size = DT_INST_REG_SIZE(inst), \
-			.inst_num = inst, \
+			.emac_index = DT_INST_PROP(inst, emac_index), \
```
Comment (MaureenHelm) [inline drivers/ethernet/eth_cyclonev.c:1154]: These changes aren't related to the commit message `Change the bit label to include the BIT() macro to tidy up the code.`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please explain the bugs you're fixing in the commit messages

File: drivers/ethernet/eth_cyclonev.c (hunk 5 above)
Comment (benkyd) [reply drivers/ethernet/eth_cyclonev.c]: Fixed in [c38c85f](https://github.com/zephyrproject-rtos/zephyr/pull/53693/commits/c38c85f8300ee6eb263cf8d2a80d886a0d58f4ce) Changes reflected in [4f2ab80](https://github.com/zephyrproject-rtos/zephyr/pull/53693/commits/4f2ab80b3d3d648acb02d5a2af3bbc876d00b964)

File: drivers/ethernet/eth_cyclonev.c (hunk 4 above)
Comment (benkyd) [reply drivers/ethernet/eth_cyclonev.c]: Added housekeeping commit [291ff11](https://github.com/zephyrproject-rtos/zephyr/pull/53693/commits/291ff11b6600c8c739216a185c2592d2972b3359)

File: dts/arm/intel_socfpga_std/socfpga.dtsi (hunk 3 above)
Comment (MaureenHelm) [reply dts/arm/intel_socfpga_std/socfpga.dtsi:148]: Please add this explanation to the commit message

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]
